vlm/src/main/scala/geotrellis/contrib/vlm/MosaicRasterSource.scala

```diff
@@ -0,0 +1,149 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package geotrellis.contrib.vlm
+
+import cats.Semigroup
+import cats.implicits._
+import cats.data.NonEmptyList
+import geotrellis.vector._
+import geotrellis.raster._
+import geotrellis.raster.merge.Implicits._
+import geotrellis.raster.resample._
+import geotrellis.raster.reproject.Reproject
+import geotrellis.proj4.{CRS, WebMercator}
+import geotrellis.raster.io.geotiff.{AutoHigherResolution, OverviewStrategy}
+import geotrellis.raster.render._
+import geotrellis.spark.tiling.LayoutDefinition
+import geotrellis.util.GetComponent
+
+/**
+  * Single threaded instance of a reader for reading windows out of collections
+  * of rasters
+  *
+  */
+trait MosaicRasterSource extends RasterSource {
+  // Orphan instance for semigroups for rasters, so we can combine
+  // Option[Raster[_]]s later
+  implicit val rasterSemigroup: Semigroup[Raster[MultibandTile]] =
+    new Semigroup[Raster[MultibandTile]] {
+      def combine(l: Raster[MultibandTile], r: Raster[MultibandTile]) = {
+        val targetRE = RasterExtent(
+          l.rasterExtent.extent combine r.rasterExtent.extent,
+          l.rasterExtent.cellSize)
+        val result = l.resample(targetRE) merge r.resample(targetRE)
+        result
+      }
+    }
+
+  /**
+    * The underlying [[RasterSource]]s that you'll use for data access
+    */
+  val sources: NonEmptyList[RasterSource]
+
+  /**
+    * The commonCrs to reproject all [[RasterSource]]s to anytime we need information about their data
+    *
+    * Since MosaicRasterSources represent collections of [[RasterSource]]s, we don't know in advance
+    * whether they'll have the same CRS. commonCrs allows specifying the CRS on read instead of
+    * having to make sure at compile time that you're threading CRSes through everywhere correctly.
+    */
+  val commonCrs: CRS
+
+  /**
+    * Uri is required only for compatibility with RasterSource.
+    *
+    * It doesn't make sense to access "the" URI for a collection, so this throws an exception.
+    */
+  def uri: String = throw new NotImplementedError(
+    """
+      | MosaicRasterSources don't have a single uri. Perhaps you wanted the uri from
+      | one of this MosaicRasterSource's sources?
+    """.trim.stripMargin
+  )
+
+  def crs: CRS = commonCrs
+
+  /**
+    * The bandCount of the first [[RasterSource]] in sources
+    *
+    * If this value is larger than the bandCount of later [[RasterSource]]s in sources,
+    * reads of all bands will fail. It is a client's responsibility to construct
+    * mosaics that can be read.
+    */
+  def bandCount: Int = sources.head.bandCount
+
+  def cellType: CellType = sources.head.cellType
```

Comment:
Probably want to get the LuB across all sources... really should be easier to do this from GT

Reply:
Is there an example of somewhere where we do this? My cell type experience so far has been a huge headache

Reply:
There is not a perfect way to handle this. [`CellType.union`](https://github.com/locationtech/geotrellis/blob/master/raster/src/main/scala/geotrellis/raster/CellType.scala#L73) is the closest thing, but will not resolve any differences in `NODATA` values.